File: 백준/Silver/1654. 랜선 자르기/랜선 자르기.py
import sys

input=sys.stdin.readline


K, N = map(int, input().split())

total = 0
stick = list()
for _ in range(K):

    cm = int(input())
    stick.append(cm)

start, end = 1, max(stick)

while start <= end:

    mid = (start+end) // 2
    # print(start, end, mid)
    count = 0
    for i in stick:
        count += i // mid
    # print(count)


    if count < N:
        end = mid - 1
    else:
        start = mid + 1

print(end)

File: 백준/Silver/1654. 랜선 자르기/랜선 자르기.js
let fs = require('fs')
let input = fs.readFileSync("/dev/stdin").toString().trim().split("\n")
// let input = fs.readFileSync("./EX.txt").toString().trim().split("\n")

const N = parseInt(input[0].split(" ")[0])
const K = parseInt(input[0].split(" ")[1])

const arr = input.slice(1, ).map(x => parseInt(x))

let start = 1
let end = Math.max(...arr)
let result = 0
while (start <= end) {
    let cnt = 0
    let mid = Math.trunc((start + end) / 2)
    for (let i = 0; i < arr.length; i++) {
        cnt += Math.trunc(arr[i] / mid)
    }
    if (cnt < K) {
        end = mid - 1
    } else {
        start = mid + 1
    }
}
console.log(end)